packages/aws-cdk/lib/api/util/cloudformation/stack-activity-monitor.ts

```diff
@@ -475,6 +479,7 @@ export class HistoryActivityPrinter extends ActivityPrinterBase {
   public addActivity(activity: StackActivity) {
     super.addActivity(activity);
     this.printable.push(activity);
+    this.print();
```

Comment:
This could have went into the the `stop()` method however the progress output would not be as accurate.

For example if moved to `stop()` you would see batches of activities being reported with the most recent progress:

<img width="931" alt="CleanShot 2021-06-21 at 17 11 02@2x" src="https://user-images.githubusercontent.com/17558268/122834150-ae4a0e80-d2b3-11eb-8f3e-eca651e40b78.png">

Instead I moved this to the `addActivity()` so that progress would correlate with the incrementing event:

<img width="918" alt="CleanShot 2021-06-21 at 17 02 52@2x" src="https://user-images.githubusercontent.com/17558268/122833477-a8076280-d2b2-11eb-8d39-53ac673ebb97.png">

_Notice how the progress updates with the event_

Reply:
I don't think I see the difference between these screenshots? Can you point it out to me?

Reply:
Sorry about that, the difference is easy to miss.

**Before**: _Notice how the highlighted area has 2 `_COMPLETE` events but the progress jumped from `2/4` to `4/4` and both events are labeled `4/4`. This is because we print to the console in chunks rather than on each event._

<img width="931" alt="before" src="https://user-images.githubusercontent.com/17558268/124959717-20736080-dfe1-11eb-9a4c-5732b06e6d59.png">


**After**: _Notice how the highlighted area has the same 2 `_COMPLETED` events, however this time we are printing to the console immediately after the event is received. This is why we see the progress increment `3/4` -> `4/4` This provides a more accurate and responsive representation of progress._

<img width="918" alt="after" src="https://user-images.githubusercontent.com/17558268/124959723-223d2400-dfe1-11eb-8d6b-fb704e418dab.png">